packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -958,6 +964,13 @@ export interface SubnetConfiguration {
    * @default false
    */
   readonly reserved?: boolean;
+
+  /**
+   * Controls if a public IP is associated to an instance at launch
+   *
+   * @default true in Subnet.Public, false in Subnet.Private or Subnet.Isolated.
```

Comment:
Should have a comment to say it ONLY works if subnetType == PUBLIC.

Should have a validation that it's not set to true if subnetType != PUBLIC.

Reply:
Sorry I was on leave and did not see it earlier. Added the comments, makes sense. You mentioned something about a test, but I am unsure on how to add that.